ui/lib/core/addon/components/text-file.js

```diff
@@ -0,0 +1,68 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+
+/**
+ * @module TextFile
+ * `TextFile` components render a file upload input with the option to toggle a "Enter as text" button
+ *  that changes the input into a textarea
+ *
+ * @example
+ * <TextFile
+ *  @hideTextAreaToggle={{true}}
+ *  @helpText="help text"
+ *  @onChange={{this.handleChange}}
+ *  @label="PEM Bundle"
+ * />
+ *
+ * @param {function} onChange - Callback function to call when the value of the input changes, returns an object in the shape of { value: fileContents, filename: 'some-file.txt' }
+ * @param {bool} [hideTextAreaToggle=false] - When true, renders a static file upload input and removes the option to toggle and input plain text
+ * @param {string} [helpText] - Text underneath label.
+ * @param {string} [label='File']  - Text to use as the label for the file input. If none, default of 'File' is rendered
+ */
+
+export default class TextFileComponent extends Component {
+  @tracked content = '';
+  @tracked filename = '';
+  @tracked uploadError = '';
+  @tracked showValue = false;
+  @tracked showTextArea = false;
+
+  async readFile(file) {
```

Comment:
refactored to no longer use `FileReader.readAsText()` after reviewing [docs](https://developer.mozilla.org/en-US/docs/Web/API/FileReader/readAsText)